extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigServerClientConfigSourceProvider.java

```diff
@@ -28,23 +28,27 @@ public SpringCloudConfigServerClientConfigSourceProvider(SpringCloudConfigClient
         this.applicationName = applicationName;
         this.activeProfile = activeProfile;
 
-        springCloudConfigClientGateway = new DefaultSpringCloudConfigClientGateway(springCloudConfigClientConfig);
+        springCloudConfigClientGateway = new VertxSpringCloudConfigGateway(springCloudConfigClientConfig);
     }
 
     @Override
     public Iterable<ConfigSource> getConfigSources(ClassLoader forClassLoader) {
+        List<ConfigSource> result = new ArrayList<>();
         try {
-            final Response response = springCloudConfigClientGateway.exchange(applicationName, activeProfile);
-            final List<Response.PropertySource> propertySources = response.getPropertySources();
-            Collections.reverse(propertySources); // reverse the property sources so we can increment the ordinal from lower priority to higher
-            final List<ConfigSource> result = new ArrayList<>(propertySources.size());
-            for (int i = 0; i < propertySources.size(); i++) {
-                final Response.PropertySource propertySource = propertySources.get(i);
-                // Property sources obtained from Spring Cloud Config are expected to have a higher priority than even system properties
-                // 400 is the ordinal of SysPropConfigSource, so we use 450 here
-                result.add(new InMemoryConfigSource(450 + i, propertySource.getName(), propertySource.getSource()));
+            Response response = springCloudConfigClientGateway.exchange(applicationName, activeProfile).await()
+                    .atMost(springCloudConfigClientConfig.connectionTimeout
```

Comment:
As discussed in chat, we need to use `.await().indefinitely()`  when `connectionTimeout` and `readTimeout` are `0`